src/py/mat3ra/made/tools/build/interface/builders.py

```diff
@@ -102,11 +108,16 @@ def _generate(self, configuration: InterfaceBuilder._ConfigurationType) -> List[
             configuration.film_configuration, configuration.film_termination
         )
         substrate_slab_ase = self.__preprocess_slab_configuration(
-            configuration.substrate_configuration, configuration.substrate_termination
+            configuration.substrate_configuration,
+            configuration.substrate_termination,
+            slab_as_is=self.build_parameters.slab_as_is,
         )
 
         if self.build_parameters.scale_film:
-            film_slab_ase.set_cell(substrate_slab_ase.cell, scale_atoms=True)
+            temp_cell = [substrate_slab_ase.cell[0], substrate_slab_ase.cell[1], film_slab_ase.cell[2]]
+            # Add z of previous height
```

Comment:
Comment should say: "We scale the film in XY direction only, and use two scaling steps and a temporary cell"